### DISCUSSION TITLE
Set filetype to be text for new files

### DISCUSSION THREAD

**Discussion by Irubataru** [LazyVim/LazyVim#4346] [Q&A] [answered]
I want to set the default filetype to be `text` for new files that doesn't have a filename. E.g. what you get by pressing <kbd>n</kbd> on the default LazyVim setup, or just by running `:new`. I tried the following autocmd

```lua
vim.api.nvim_create_autocmd("BufEnter", {
  group = vim.api.nvim_create_augroup("empty_filetype", { clear = true }),
  callback = function()
    if vim.bo.filetype == "" then
      vim.bo.filetype = "text"
    end
  end,
})
```

It seems to work, however, when I now open telescope, I get a red squiggly line under the prompt icon.

![image](https://github.com/user-attachments/assets/a45e0e5d-e3a4-4d10-80cb-e940befbd31d)

And hovering over it shows this

![image](https://github.com/user-attachments/assets/809096fd-8301-4ba8-9a50-ff86aba18eaa)

Any idea what's going on or how I should do this?

**Comment by dpetka2001** [comment, marked as answer]
It seems that when the autocmd happens at the `BufEnter` event, the Telescope buffer hasn't set its filetype yet. Therefore it thinks it's text filetype and enables spelling I believe (which you can check with `:set spell?`). If you delay the detection of the filetype for a bit like so
```lua
vim.api.nvim_create_autocmd("BufEnter", {
  group = vim.api.nvim_create_augroup("empty_filetype", { clear = true }),
  callback = function()
    vim.schedule(function()
      if vim.bo.filetype == "" then
        vim.bo.filetype = "text"
      end
    end)
  end,
})
```
You  won't see the red undercurl under the prompt icon any more.

**Reply by Irubataru** [reply]
This worked like a charm, thank you so much 😊
